File: py/66. Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        if digits[-1] == 9:
            i = len(digits) - 1
            while i >= 0:
                if digits[i] + 1 > 9:
                    digits[i] = 0
                    i -= 1
                else:
                    digits[i] += 1
                    break
            if i == -1:
                digits.insert(0, 1)
        else:
            digits[-1] += 1
        return digits

File: cpp/66_Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digit) {
        
        int carry =1;
        
        for(int i=digit.size()-1;i>=0;i--)
        {
            int k = digit[i]+carry;
             digit[i] = k%10;
            carry = k/10;
            
        }
        if(digit[0]==0)
        {
            digit.insert(digit.begin(),1);
        }
        
        
        return digit;
        
        
    }
};
